src/pages/index.js

```diff
@@ -0,0 +1,346 @@
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardRemoveBtn = cardElement.querySelector(".card__remove-btn");
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+    cardElement.classList.add("card_liked");
+  }
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalCaptionEl.textContent = data.name;
+    previewModalImageEl.alt = data.name;
+  });
+
+  cardLikeBtn.addEventListener("click", (evt) => {
+    handleLike(evt, data._id);
+  });
+
+  cardRemoveBtn.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+  //cardRemoveBtn.classList.toggle("card__remove-btn_hovered");
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+
+function handleEscapeKey(event) {
+  if (event.key === "Escape") {
+    const openModal = document.querySelector(".modal_opened"); // Find the currently open modal
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-btn_liked");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then((updatedData) => {
+      const updatedIsLiked = updatedData.likes && updatedData.likes.length > 0;
+
+      if (updatedData.isLiked) {
+        evt.target.classList.add("card__like-btn_liked");
+      } else {
+        evt.target.classList.remove("card__like-btn_liked");
+      }
+    })
+    .catch(console.error);
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileDescriptionElement.textContent = data.about;
+      evt.target.reset();
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  const avatar = avatarInput.value;
+  api
+    .editAvatar(avatar)
+    .then((userData) => {
+      document.querySelector(".profile__avatar").src = userData.avatar;
```

Comment:
`document.querySelector(".profile__avatar")`



It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine